electionleaflets/settings/base.py

```diff
@@ -1,32 +1,29 @@
-import sys
-from os.path import join, abspath, dirname
+import environ, sys
+root = environ.Path(__file__) - 2 # three folder back (/a/b/c/ - 3 = /)
+env = environ.Env() # set default values and casting
 
-# PATH vars
-here = lambda *x: join(abspath(dirname(__file__)), *x)
-PROJECT_ROOT = here("..")
-root = lambda *x: join(abspath(PROJECT_ROOT), *x)
 sys.path.insert(0, root('third_party'))
 sys.path.insert(0, root('apps'))
 sys.path.insert(0, '../django-uk-political-parties/')
 
+DEBUG = env.bool('DEBUG', default=False)
+TEMPLATE_DEBUG = DEBUG
 
-DEBUG = False
-template_DEBUG = DEBUG
-
-# DATABASES define in environment specific settings file
+db_url = env('DATABASE_URL', default='postgres://postgres@localhost/electionleaflets')
```

Comment:
Ok, this comment on the general change to use env vars in this file.

Absolutely agree with 12fa, but I think I would rather do it in a way that's similar to our other apps. That is, having defaults in this file, overwritten by a `local.py` in the same directory.  

Lot of "settings" are actually app-wide "constants" that should be in version control. For these we can adopt the sort of constants approach we've used on WDIV: https://github.com/DemocracyClub/UK-Polling-Stations/tree/master/polling_stations/settings

I also feel that writing settings to a file has a couple of advantages:

1. It makes debugging easier, because you know you can read settings from a file rather than trying to figure out the gunicorn environment that's _actually_ running the app
2. It makes local dev easier by not having to set env vars (I know the `.env` file solves this, but then we're back at just adding settings to `local.py`
3. Editing the `.env` file means changes don't get picked up by django dev server's auto reloading. This is a small point, but one that can cause problems if you forget about it!